File: LRUCache.js
class LRUCache{
    constructor(capacity){
        this.capacity=capacity
        this.cache={}
    }
    put(key, value){
        console.log(Object.keys(this.cache).length)
        if (key in this.cache){
            delete this.cache[key]
        }
        else if(Object.keys(this.cache).length===this.capacity){
            delete this.cache[Object.keys(this.cache)[0]]   
        }
        this.cache[key]=value
    }
    get(key){
        if (key in this.cache){
            let value = this.cache[key] //40
            delete this.cache[key]
            this.cache[key]=value
            return value
        }
        else{
            return -1
        }
    }
}

obj=new LRUCache(4)
obj.put('c', 9)

console.log(obj.cache)
console.log(obj.get('f'))
console.log(obj.cache)

File: LRUCache.py
import collections
class LRUCache:
    def __init__(self,capacity):
        self.capacity=capacity
        self.cache=collections.OrderedDict()
    def put(self,key,value):
        if key in self.cache:
            self.cache.pop(key)
        elif len(self.cache)==self.capacity:
            self.cache.popitem(last=False)
        self.cache[key]=value
    def get(self, key):
        if not key in self.cache:
            return -1
        value=self.cache.pop(key)
        print(value)
        self.cache[key]=value
        return value

obj=LRUCache(4)
obj.put('a',4)
obj.put('b',3)
obj.put('c',9)
obj.put('b',7)
obj.put('d',8)
obj.put('e',6)
print(obj.cache)
print(obj.get('d'))
    